sqlmesh/core/model/meta.py

```diff
@@ -109,13 +109,17 @@ def _partition_by_validator(
                 d.parse_one(entry, dialect=dialect) if isinstance(entry, str) else entry
                 for entry in ensure_list(v)
             ]
-        partitions = [
-            exp.to_column(expr.name) if isinstance(expr, exp.Identifier) else expr
-            for expr in partitions
-        ]
 
-        for partition in partitions:
-            partition.meta["dialect"] = dialect
+        validated_partitions = []
+        for expr in partitions:
```

Comment:
i don't think it's as important to do it for this section or the types if that will simplify the code

Reply:
The reason I modified this section is that we can have dialect-specific quotes in identifiers (not sure if it's possible to also have other dialect-specific expressions), and so some tests failed without these changes. For example, in `test_model::test_model_normalization` we have ```partitioned_by foo(`ds`) ```, so the following assertion failed:

```python
assert model.partitioned_by[0].sql(dialect="bigquery") == "foo(`ds`)"
```

Reply:
it shouldn't fail, i think the issue may be that you're not serializing in the correct dialect?